src/main/scala/is/hail/expr/AST.scala

```diff
@@ -236,7 +236,7 @@ sealed abstract class AST(pos: Position, subexprs: Array[AST] = Array.empty) {
   def runAggregator(ec: EvalContext): CPS[Any] = {
     val typedNames = ec.st.toSeq
       .sortBy { case (_, (i, _)) => i }
-      .map { case (name, (_, typ)) => (name, typ) }
+      .map { case (name, (_, typ)) => (name, typ.scalaClassTag.asInstanceOf[ClassTag[AnyRef]]) }
```

Comment:
Yes, you're absolutely correct, I somehow forgot how this worked.

Let me try again, I think some issue came up because I wanted this:

```scala
Map[String, (ClassTag[T], Code[T]) forSome { type T }]
```